AFC.py

```diff
@@ -647,30 +648,27 @@ def TOOL_UNLOAD(self, CUR_LANE):
 
     cmd_CHANGE_TOOL_help = "change filaments in tool head"
     def cmd_CHANGE_TOOL(self, gcmd):
-        #self.toolhead = self.printer.lookup_object('toolhead')
         lane = gcmd.get('LANE', None)
+        self.failed_in_toolchange = False
         if lane != self.current:
-            store_pos = self.toolhead.get_position()
-            if self.is_printing() and not self.is_paused():
-                self.change_tool_pos = store_pos
+            # Create save state
+            self.gcode.run_script_from_command("SAVE_GCODE_STATE NAME=_AFC_CHANGE_TOOL")
             if self.current != None:
                 CUR_LANE = self.printer.lookup_object('AFC_stepper ' + self.current)
                 self.TOOL_UNLOAD(CUR_LANE)
             CUR_LANE = self.printer.lookup_object('AFC_stepper ' + lane)
             self.TOOL_LOAD(CUR_LANE)
-            newpos = self.toolhead.get_position()
-            newpos[2] = store_pos[2]
-            self.toolhead.manual_move(newpos, self.tool_unload_speed)
-            self.toolhead.wait_moves()
-            if self.is_printing() and not self.is_paused():
-                self.change_tool_pos = None
+            # Restore state
+            self.gcode.run_script_from_command("RESTORE_GCODE_STATE NAME=_AFC_CHANGE_TOOL MOVE=1 MOVE_SPEED={}".format(self.tool_unload_speed))
```

Comment:
I gotta step away for a bit, but I want to also set up a different move speed here. Our default for this is pretty slow (25mm/s) which leads to excess oozing while restoring position after wiping.

Reply:
Might be a good idea to make the move speed customizable.